barman/server.py

```diff
@@ -1516,47 +1516,85 @@ def delete_backup(self, backup):
         """Deletes a backup
```

Comment:
I'd say to describe in this docstring what this method does so we can get the difference between `delete_backup` and `perform_delete_backup` in the future, if needed.

I mean, something similar to what you did in your commit, which is clearer.